### DISCUSSION TITLE
tensor.squeeze() problem

### DISCUSSION THREAD

**Discussion by Istiak665** [mrdbourke/pytorch-deep-learning#261] [Q&A] [answered]
I am trying to remove extra dimension from the mentioned tensor. But squeeze() is not working. For your kind information my current device ion "cpu". So can anyone tell me why it is not working?

![image](https://user-images.githubusercontent.com/72601207/212610451-2959464d-7423-4301-a746-90fd5b61d412.png)

**Comment by ryan-popa** [comment, marked as answer]
squeeze only removes 1 dimension shapes. You shape is 5,4 - there is no 1 size dim to be removed

<img width="408" alt="image" src="https://user-images.githubusercontent.com/6673637/212638263-e1de7b31-cbd3-4eef-8e2a-a27eb61f7c69.png">

**Reply by Istiak665** [reply]
Yes, now I understood. Thank you so much.

**Reply by mrdbourke** [reply]
@ryan-popa is correct, thank you for sharing/helping out!

**Comment by mfarooq33** [comment]
Thank you Rayn Popa. Now its clear.
